src/biome/text/dataset_readers/sequence_classifier_dataset_reader.py

```diff
@@ -103,60 +85,37 @@ def _read(self, file_path: str) -> Iterable[Instance]:
         return (instance for idx, instance in instances.iteritems() if instance)
 
     def text_to_instance(
-        self, text: Union[Dict, "dask.Series", "pandas.Series"]
+        self,
+        tokens: Union[str, List[str], dict],
+        label: Optional[str] = None,
+        # TODO we can use the extra args for metadata build. Not used for now but required
+        **extra_args
     ) -> Optional[Instance]:
         """Extracts the forward parameters from the example and transforms them to an `Instance`
 
         Parameters
         ----------
-        text
-            The keys of this dict should contain the parameter names of the `forward` method of your model.
+        tokens
+            The input tokens key,values (or the text string)
+        label
+            The label value
 
         Returns
         -------
         instance
-            Returns `None` if the example could not be transformed to an Instance.
+            Returns `None` if cannot generate an new Instance.
         """
         fields = {}
-        for param_name, param in self.forward_params.items():
-            try:
-                value = text[param_name]
-                fields[param_name] = self._value_to_field(param_name, value)
-
-            except KeyError as e:
-                # if parameter is required by the forward method raise a meaningful error
-                if param.default is Parameter.empty:
-                    raise KeyError(
-                        f"{e}; You are probably missing '{param_name}' in your forward definition of the data source."
-                    )
-
-        fields = {k: v for k, v in fields.items() if v}
-        return Instance(fields) if fields and len(fields) > 0 else None
-
-    def _value_to_field(
-        self, param_name: str, value: Any
-    ) -> Optional[Union[LabelField, TextField, ListField]]:
-        """Embeds the value in one of the `allennlp.data.fields`.
-        For now the field type is basically inferred from the hardcoded label tag ...
 
-        Parameters
-        ----------
-        param_name
-            Must match one of the parameters in the `forward` method of your model.
-        value
-            Value of the field.
+        tokens_field = self.build_textfield(tokens)
+        label_field = LabelField(label) if label else None
 
-        Returns
-        -------
-        Returns either a `LabelField` or a `TextField`/`ListField` depending on the `param_name`.
-        """
-        if not value:
-            return None
+        if tokens_field:
+            fields["tokens"] = tokens_field
+        if label_field:
+            fields["label"] = label_field
 
-        if param_name == self.LABEL_TAG:
-            return LabelField(value)
-        else:
-            return self.build_textfield(value)
+        return Instance(fields) if fields and len(fields) > 0 else None
```

Comment:
I think this is a duplicated check...
a non-empty dict will always evaluate to true and the length of an non-empty dict is always greater than 0.